### ISSUE TITLE
preserve video?

### ISSUE THREAD

**Issue by chovy** [mozilla/readability#657]
I'm seeing some markup that contain youtube links....but they don't render. All I get is an image thumbnail for youtube.

```
<div data-src="https://www.youtube.com/embed/ELoyRfcksP0" data-id="ELoyRfcksP0" data-query="version=3&amp;rel=1&amp;showsearch=0&amp;showinfo=1&amp;iv_load_policy=1&amp;fs=1&amp;hl=en-GB&amp;autohide=2&amp;wmode=transparent"><div data-id="ELoyRfcksP0" data-query="version=3&amp;rel=1&amp;showsearch=0&amp;showinfo=1&amp;iv_load_policy=1&amp;fs=1&amp;hl=en-GB&amp;autohide=2&amp;wmode=transparent" data-src="https://www.youtube.com/embed/ELoyRfcksP0"><p><img src="https://i.ytimg.com/vi/ID/hqdefault.jpg" alt="" width="480" height="360" data-lazy-src="https://i.ytimg.com/vi/ELoyRfcksP0/hqdefault.jpg"></p></div></div>
```

Is there anyway to render the videos?

**Comment by gijsk** [comment]
I'm not sure I understand - the markup you provided doesn't contain a video as such. It's possible that the site has a script that transforms this markup into a video or iframe (which then contains youtube content which has a video), but as-is this isn't a video. Can you elaborate on what you're asking for?

**Comment by chovy** [comment]
Yeah it has a video as an attribute. I was just trying to determine if this was put there @reader or if its just garbage from the source html.

**Comment by gijsk** [comment]
Difficult to say without seeing the source HTML. :-)

Can you link to the page where you saw this?

**Comment by chovy** [comment]
unfortunately not, sorry.

**Comment by gijsk** [comment]
OK, without more details it's difficult to do something here - we could potentially do something proactive here but we'd need to know more about the markup and script that the originating site uses.

**Comment by chovy** [comment]
I just see a lot of <div data-src="url"> tags.

**Comment by chovy** [comment]
Here's an eample

https://briskreader.com/read?link=https://tipjarr.net/post/Intro-to-react-comp-state-and-props

this instance its an image but I see <img data-src="url">

**Comment by gijsk** [comment]
@chovy so in this case, the website fetches the HTML itself and then uses readability? It's a bit hard to see what is going on - are you the developer/admin of this site? I see this markup in the "readerized" copy:

```html
<img alt="desert" data-srcset="https://images.unsplash.com/photo-1607753724987-7277196eac5d?ixid=MXwxMjA3fDB8MHxwaG90by1wYWdlfHx8fGVufDB8fHw%3D&amp;ixlib=rb-1.2.1&amp;auto=format&amp;fit=crop&amp;w=1267&amp;q=80" src="https://tipjarr.net/">
```

It looks like the original markup is:

```html
<img class="lazyload blur-up w-full h-48 lg:h-84 object-cover my-2 block shadow-lg" alt="desert" data-srcset="https://images.unsplash.com/photo-1607753724987-7277196eac5d?ixid=MXwxMjA3fDB8MHxwaG90by1wYWdlfHx8fGVufDB8fHw%3D&amp;ixlib=rb-1.2.1&amp;auto=format&amp;fit=crop&amp;w=1267&amp;q=80"/>
```

note that it has no `src` attribute at all. If I run this example with readability locally, I also get an `img` without a `src` attribute. I imagine that to fix this, we'd need to use the `data-srcset` attribute and translate it to just `srcset`. We've mostly fixed these with e.g. #542 but it looks like in this case, the filtering there might not apply because the URL used doesn't look like an image. We could look at de-restricting that for the specific `data-srcset` and `data-src` attributes on `img` tags, I guess, but that may be tricky. It also won't help with the video example you cited. I can reopen this ticket and morph it, I guess...

**Comment by chovy** [comment]
yeah. it just seems pretty commoon. I don't know if its a lazy laoding technique or not but I see `<div data-videosrc>` and `<img data-imgsrc>` would think I'll have to post-process the HTML myself if its not the responsibility of mozilla reader.
